src/scheduler/MpiWorld.cpp

```diff
@@ -305,46 +307,40 @@ int MpiWorld::isend(int sendRank,
                     int count,
                     faabric::MPIMessage::MPIMessageType messageType)
 {
-    return doISendRecv(
-      sendRank, recvRank, buffer, nullptr, dataType, count, messageType);
+    int requestId = (int)faabric::util::generateGid();
+
+    asyncThreadMap.emplace(
```

Comment:
The previous version used `insert` instead of `emplace` both will silently fail if the key already exists, but the `[]` operator would silently override the value if the key was repeated.

To be extra cautious we'd check that the key is indeed not there, but I think trusting `util::generateGid()` is a safe assumption (for the moment).